src/types/networks.ts

```diff
@@ -50,6 +50,10 @@ export interface IAdapterNetworkSetting<T = string> {
    */
   testId?: string;
   required?: boolean;
+  /**
+   * Can be used to override or add field validation rules
+   */
+  validationRules?: object;
```

Comment:
I rather suggest to avoid setting `object` as a type because actually most of stuff in JS is an object :) In most cases `Dictionary` is the best fit in that cases. Or some kind of `Record`.